generator/generator.py

```diff
@@ -13,14 +13,14 @@
 
 
 # handle firmware name
-def get_firmware_name(fqbn):
-    if fqbn == "arduino:megaavr:uno2018":
-        return "NINA_W102-arduino.megaavr.uno2018.bin"
-    elif fqbn == "arduino:mbed_nano:nanorp2040connect":
-        return "NINA_W102-arduino.mbed_nano.nanorp2040connect.bin"
-    elif fqbn == "arduino:samd:mkr1000":
-        return "m2m_aio_3a0-arduino.samd.mkr1000.bin"
-    return "NINA_W102.bin"
+def get_firmware_file(module, simple_fqbn, version):
+    file = Path(f'{module}-{simple_fqbn}.bin')
+    default_file = Path(f'{module}.bin')
+    firmware_path = f"firmwares/{module}/{version}/{file}"
+    firmware_full_path = Path(__file__).parent / ".." / firmware_path
+    if firmware_full_path.exists():
+        return firmware_full_path
+    return Path(str(firmware_full_path).replace(str(file), str(default_file)))
```

Comment:
This could be simplified a bit, also to avoid the same Windows issue below.

```suggestion
def get_firmware_file(module, simple_fqbn, version):
    firmware_full_path = Path(__file__).parent.parent / "firmwares" / module / version
    fqbn_specific_file_name = f"{module}-{simple_fqbn}.bin"
    if (firmware_file := firmware_full_path / fqbn_specific_file_name).exists():
        return firmware_file
    return firmware_full_path / f"{module}.bin"```